src/visualization/Viewer.js

```diff
@@ -48,10 +50,17 @@ ROS3D.Viewer = function(options) {
   var lineTypePanAndZoomFrame = options.lineTypePanAndZoomFrame || 'full';
 
   // create the canvas to render to
-  this.renderer = new THREE.WebGLRenderer({
-    antialias : antialias,
-    alpha: true
-  });
+  this.renderer = canvas ? 
+    new THREE.WebGLRenderer({
+      antialias : antialias,
+      alpha: true,
+      canvas: canvas
+    }) 
+    : 
+    new THREE.WebGLRenderer({
+      antialias : antialias,
+      alpha: true,
+    });
```

Comment:
```suggestion
  this.renderer = new THREE.WebGLRenderer({
    antialias : antialias,
    alpha: true,
    canvas: canvas
  });
```

In case `options.canvas` is not provided, `canvas` will be undefined. Which Three will check for.